pkg/ee/validation/machine/resourcequota.go

```diff
@@ -41,54 +40,9 @@ import (
 // ValidateQuota validates if the requested Machine resource consumption fits in the quota of the clusters project.
 func ValidateQuota(ctx context.Context, log *zap.SugaredLogger, seedClient, userClient ctrlruntimeclient.Client,
 	machine *clusterv1alpha1.Machine, caBundle *certificates.CABundle) error {
-	config, err := types.GetConfig(machine.Spec.ProviderSpec)
+	quotaReq, err := GetMachineResourceUsage(ctx, userClient, machine, caBundle)
```

Comment:
nit:
```suggestion
	quotaUsage, err := GetMachineResourceUsage(ctx, userClient, machine, caBundle)
```

Reply:
renamed it to `machineResourceUsage`, i feel it makes the most sense